docs/manual/config/hooks.rst

```diff
@@ -62,7 +62,7 @@ We can then subscribe to ``startup_once`` to run this script:
     @hook.subscribe.startup_once
     def autostart():
         home = os.path.expanduser('~/.config/qtile/autostart.sh')
-        subprocess.call([home])
+        subprocess.run([home], shell=True)
```

Comment:
If script contains specific shell features

Reply:
Ok. Should we add a comment about that? Using `shell=True` is generally discouraged except where actually required (such as the example you gave).

Reply:
I can add a comment with link about security considerations

Reply:
I don't think that's quite right - if the script has a shell shebang then it will always be run with a shell. `shell=True` here is used for the command executed by `subprocess.run`, for example `subprocess.run("ls > ~/files")`, which would otherwise fail saying `"No such command: 'ls > ~/files'"`.

Reply:
@m-col Oh, yes, u r right

Reply:
Ok. So we do remove it then?

Reply:
Yeah I don't think it's needed here

Reply:
But script may not contain a shebang

Reply:
I think it will error anyway, because `shell=True` doesn't apply to the executable being executed.

Reply:
ok, removed